core/src/main/java/de/danoeh/antennapod/core/util/comparator/SearchResultDateComparator.java

```diff
@@ -0,0 +1,19 @@
+package de.danoeh.antennapod.core.util.comparator;
+
+import java.util.Comparator;
+
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.feed.SearchResult;
+
+public class SearchResultDateComparator implements Comparator<SearchResult> {
+    /**
+     * Compare items and sort it on chronological order.
+     */
+    @Override
+    public int compare(SearchResult o1, SearchResult o2) {
+        if (o1.getComponent() instanceof FeedItem && o2.getComponent() instanceof FeedItem) {
```

Comment:
just some opinionated advice:
I'd invert the if (Alt/Option+Enter on macOS) - see https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

to make sure it is obvious that we sort from latest to earliest, I would write `-1 * o1....compareTo(...o2...)`
You could even name the class `InReverseChronologicalOrder` or something similar which would read nicely.

Reply:
Thanks for you suggestion. I changed it as you prescribed.